GolosChain/golos.contracts
# Disable vesting and storables in charge contract #835

Contracts:
+ Add 2 defines to charge contract to disable vesting and storables
+ charge `use` action now fails if non-zero `vesting_price` passed
+ charge `setrestorer` action now fails if function contains `"v"` variable
+ posting `createmssg` action now fails if `vestpayment==true`
+ posting `setlimit` action now fails if non-zero `vesting_price` or `min_vesting` passed
+ charge `useandstore` and `removestored` actions not used. their implementation removed if `DISABLE_CHARGE_STORABLE` defined. Note: actions and related tables are still in abi (can't conditionally enable/disable)
+ minor fixes and optimizations

Tests:
+ Add 2 defines to charge test api to disable vesting and storables tests
+ add `disabled_vesting` charge test and `disabled_charge_vesting` publication test
+ fix tests which used vesting-powered charges (`limits_test` timeout set as workaround, it should be based on something to test properly)
